### PR TITLE
setup up individual krb5cc files for each individual job.  prevents c…

### PR DESCRIPTION
**Pull request by johnyu0520**
…ollision of kerberos tickets between different jobs and processes

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 1]
```diff
@@ -81,6 +87,12 @@ public void run() throws Exception {
     info(commands.size() + " commands to execute.");
     File[] propFiles = initPropsFiles();
     Map<String, String> envVars = getEnvironmentVariables();
+    
+    // change krb5ccname env var    
+    String effectiveUser = getEffectiveUser(jobProps);
+    String krb5ccname =
+        String.format("/tmp/krb5cc__%s__%s__%s__%s__%s", jobProps.getString(CommonJobProperties.PROJECT_NAME),jobProps.getString(CommonJobProperties.FLOW_ID), jobProps.getString(CommonJobProperties.JOB_ID), jobProps.getString(CommonJobProperties.EXEC_ID), effectiveUser);
```
Comment (hluu) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java]: reformat this line

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java [hunk 2]
```diff
@@ -119,6 +131,31 @@ public void run() throws Exception {
...
+    {
+      effectiveUser = jobProps.getString(USER_TO_PROXY);
+    }
+    else if (jobProps.containsKey(CommonJobProperties.SUBMIT_USER))
+    {
+      effectiveUser = jobProps.getString(CommonJobProperties.SUBMIT_USER);
+    }
+    else
+    {
+      throw new RuntimeException("There is no user.to.proxy and submit.user in the jobProps.  There is something wrong with the system");
```
Comment (hluu) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java]: shorten the error msg

File: azkaban-common/src/test/java/azkaban/jobExecutor/ProcessJobTest.java [hunk 3]
```diff
@@ -46,6 +46,12 @@ public void setUp() throws IOException {
     props.put(AbstractProcessJob.WORKING_DIR, workingDir.getCanonicalPath());
     props.put("type", "command");
     props.put("fullPath", ".");
+    
+    props.put(CommonJobProperties.PROJECT_NAME, "test_project");
+    props.put(CommonJobProperties.FLOW_ID, "test_flow");
+    props.put(CommonJobProperties.JOB_ID, "test_job");
+    props.put(CommonJobProperties.EXEC_ID, "123");
+    props.put(CommonJobProperties.SUBMIT_USER, "test_user");
```
Comment (hluu) [inline azkaban-common/src/test/java/azkaban/jobExecutor/ProcessJobTest.java]: Add a test 
* Remove SUBMIT_USER and add USER_TO_PROXY
* Make sure job run successfully

Add a test w/o both submit_user and user_to_proxy

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java (hunk 1 above)
Comment (hluu) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java]: replace space with underscore for project name, flow name, job name. etc.

Comment (johnyu0520) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java]: done

File: azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java (hunk 2 above)
Comment (johnyu0520) [inline azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java]: done

**Comment by hluu** [comment]
LGTM

**Pull request merged**